src/InputOutput/writers.jl

```diff
@@ -42,6 +47,18 @@ struct HDF5Writer{C <: ClimaComms.AbstractCommsContext} <: AbstractWriter
     cache::Dict{String, String}
 end
 
+function HDF5Writer(
+    f::Function,
+    filename::AbstractString,
+    context::ClimaComms.AbstractCommsContext;
+    overwrite::Bool = true,
+)
+
+    writer = HDF5Writer(filename, context; overwrite)
+    f(writer)
+    Base.close(writer)
```

Comment:
I think this should also be enclosed in a `try-finally`, otherwise there is no guarentee that the file will be closed (e.g., if `f` fails)

Reply:
Ah, whoops, yes, thank you!